grpc-protocol/src/main/java/com/linecorp/armeria/common/grpc/protocol/Base64Decoder.java

```diff
@@ -0,0 +1,180 @@
+            -9, -9, -9, // Decimal 44 - 46
+            63, // Slash at decimal 47
+            52, 53, 54, 55, 56, 57, 58, 59, 60, 61, // Numbers zero through nine
+            -9, -9, -9, // Decimal 58 - 60
+            -1, // Equals sign at decimal 61
+            -9, -9, -9, // Decimal 62 - 64
+            0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, // Letters 'A' through 'N'
+            14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, // Letters 'O' through 'Z'
+            -9, -9, -9, -9, -9, -9, // Decimal 91 - 96
+            26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, // Letters 'a' through 'm'
+            39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, // Letters 'n' through 'z'
+            -9, -9, -9, -9, -9, // Decimal 123 - 127
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 128 - 140
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 141 - 153
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 154 - 166
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 167 - 179
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 180 - 192
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 193 - 205
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 206 - 218
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 219 - 231
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9,     // Decimal 232 - 244
+            -9, -9, -9, -9, -9, -9, -9, -9, -9, -9, -9              // Decimal 245 - 255
+    };
+
+    private final ByteBufAllocator allocator;
+    @Nullable
+    private ByteBuf dest;
+    private final byte[] last3 = new byte[3]; // The 4th byte is stored in a local variable.
+    private int pos;
+
+    Base64Decoder(ByteBufAllocator allocator) {
+        this.allocator = allocator;
+    }
+
+    ByteBuf decode(ByteBuf src) {
+        final ByteBuf dest = allocator.buffer(decodedBufferSize(src.readableBytes()));
+        this.dest = dest;
+        boolean success = false;
+        try {
+            src.forEachByte(this);
+            success = true;
+            return dest;
+        } finally {
+            if (!success) {
+                dest.release();
+            }
+            src.release();
+        }
+    }
+
+    private int decodedBufferSize(int len) {
+        return (len + pos) / 4 * 3;
+    }
+
+    @Override
+    public boolean process(byte value) throws Exception {
+        final byte decodedByte = DECODABET[value & 0xFF];
+        if (decodedByte < WHITE_SPACE_ENC) {
+            throw new IllegalArgumentException(
+                    "invalid Base64 input character: " + (short) (value & 0xFF) + " (decimal)");
+        }
+
+        // White space, Equals sign or better
+        if (decodedByte < EQUALS_SIGN_ENC) {
+            // Ignore the white space.
```

Comment:
Why do we ignore white space, that seems like malformed base64?

Reply:
Let me raise an exception. 😄